### PR TITLE
[Spring] Add @CucumberContextConfiguration annotation

### PR DESCRIPTION
**Pull request by rasklaad**
## Summary

To make Cucumber aware of your test configuration you can annotate a
configuration class with `@CucumberContextConfiguration` and with one of the
following annotations: `@ContextConfiguration`, `@ContextHierarchy` or
`@BootstrapWith`. If you are using SpringBoot, you can annotate configuration
class with `@SpringBootTest(classes = TestConfig.class)`.

## Details

- [x] Spring backend minimum implementation
- [x] Spring backend tests
- [x] Docs/javadocs

## Motivation and Context
Fixes: https://github.com/cucumber/cucumber-jvm/issues/1896

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [x] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by mpkorstanje** [comment]
Don't worry about `revapi` failing the build right now. It's a semantic version checker. You can ignore it, I'll adjust the version numbers.

File: spring/src/main/java/io/cucumber/spring/SpringBackend.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Override
+    public void disposeWorld() {
+
+    }
+
+    @Override
+    public Snippet getSnippet() {
+        return null;
```
Comment (mpkorstanje) [inline spring/src/main/java/io/cucumber/spring/SpringBackend.java]: We should make sure the caller in `Runner.java` filters out the null values.

File: spring/src/main/java/io/cucumber/spring/CucumberSpringTest.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@API(status = API.Status.STABLE)
+public @interface CucumberSpringTest {
```
Comment (mpkorstanje) [inline spring/src/main/java/io/cucumber/spring/CucumberSpringTest.java]: I suggested `@CucumberSpringTest` but I'm not sure if that is the right name. This annotation allows a user to mark a application context configuration class as the class that Cucumber should use. Maybe different name can convey this better.

Comment (rasklaad) [reply spring/src/main/java/io/cucumber/spring/CucumberSpringTest.java]: What about `@CucumberContextConfiguration` ?

Comment (mpkorstanje) [reply spring/src/main/java/io/cucumber/spring/CucumberSpringTest.java]: That would work.

**Comment by mpkorstanje** [comment]
And we're done. Thanks you very much!

**Pull request merged**